pkg/services/ngalert/api/api_provisioning.go

```diff
@@ -589,11 +590,16 @@ func exportHcl(download bool, body definitions.AlertingFileExport) response.Resp
 	convertToResources := func() error {
 		for idx, group := range body.Groups {
 			gr := group
-			resources = append(resources, hcl.Resource{
-				Type: "grafana_rule_group",
-				Name: fmt.Sprintf("rule_group_%04d", idx),
-				Body: &gr,
-			})
+			toHashedGruopIdx := append([]byte(gr.Name), []byte(string(idx))...)
+			hash, err := fnv.New32().Write(toHashedGruopIdx)
+			if err != nil {
+				resources = append(resources, hcl.Resource{
+					Type: "grafana_rule_group",
+					Name: fmt.Sprintf("rule_group_%04d", hash),
```

Comment:
I think if we format it to hex , it will be more clearer that this is a hash.
Let's format as `%016x`